frontend_tests/puppeteer_tests/settings.ts

```diff
@@ -208,7 +208,13 @@ async function test_invalid_edit_bot_form(page: Page): Promise<void> {
         await common.get_text_from_selector(page, "div.bot_edit_errors"),
         "Name is already in use!",
     );
-    await page.click("#edit_bot_modal .dialog_cancel_button");
+
+    const cancel_button_selector = "#edit_bot_modal .dialog_cancel_button";
+    await page.waitForFunction(
+        (cancel_button_selector: string) =>
+            !document.querySelector(cancel_button_selector)?.hasAttribute("disabled"),
+    );
```

Comment:
This `waitForFunction` call is missing an argument. Perhaps that’s why it didn’t work (#20815).

```diff
     await page.waitForFunction(
         (cancel_button_selector: string) =>
             !document.querySelector(cancel_button_selector)?.hasAttribute("disabled"),
+            cancel_button_selector,
     );
```